pandas/core/indexing.py

```diff
@@ -516,8 +519,10 @@ def can_do_equal_len():
             self.obj._data = self.obj._data.setitem(indexer=indexer, value=value)
             self.obj._maybe_update_cacher(clear=True)
 
-    def _align_series(self, indexer, ser):
+    def _align_series(self, indexer, ser, indexer_multiindex=False):
         # indexer to assign Series can be tuple, slice, scalar
+        # indexer_multiindex should be set to True if the indexer came from
+        # a multi index. Otherwise, the items will be broadcast unnecessarily
         if isinstance(indexer, (slice, np.ndarray, list, Index)):
             indexer = tuple([indexer])
```

Comment:
same here.

can you update the doc-string as well (e.g. with a ``Parameters/Returns`` section)